core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -132,6 +110,7 @@ class TopicDeletionManager(controller: KafkaController,
   def enqueueTopicsForDeletion(topics: Set[String]) {
     if (isDeleteTopicEnabled) {
       topicsToBeDeleted ++= topics
+      topics.foreach(controller.partitionStateMachine.excludeDeletingTopicFromOfflinePartitionCount)
```

Comment:
Could this be called multiple times for the same topic?  the enqueue event is trigger whenever there is a topic deletion.  user can issue multiple topic deletions in parallel.  If this is the case, it can result in undercount of offline partition.

Reply:
Good catch. We should only update the counts for the new topics to be deleted. Updated the PR to address this.